docker/docker_client.go

```diff
@@ -544,10 +548,21 @@ func (c *dockerClient) setupRequestAuth(req *http.Request, extraScope *authScope
 				token = t.(bearerToken)
 			}
 			if !inCache || time.Now().After(token.expirationTime) {
-				t, err := c.getBearerToken(req.Context(), challenge, scopes)
+				var (
+					t   *bearerToken
+					err error
+				)
+
+				if c.auth.IdentityToken != "" {
+					t, err = c.getBearerTokenOAuth2(req.Context(), challenge, scopes)
+				} else {
+					t, err = c.getBearerToken(req.Context(), challenge, scopes)
+				}
+
```

Comment:
(Absolutely non-blocking: Maybe drop this line to keep the error check visually close.)